common/client-core/src/client/mix_traffic/mod.rs

```diff
@@ -111,7 +152,26 @@ impl MixTrafficController {
                 }
             }
             shutdown.recv_timeout().await;
+
+            if self.forget_me {
+                log::info!("Sending forget me request to the gateway");
+                match self
+                    .gateway_transceiver
+                    .send_client_request(ClientRequest::ForgetMe {
```

Comment:
have you checked it actually works? because this is received after the global shutdown has already been issued so it might be the case that the gateway client might not work anymore because some of the channels it was sending received messages on are already gone